rs/rust_canisters/random_traffic_test/src/main.rs

```diff
@@ -0,0 +1,236 @@
+use candid::{Decode, Encode};
+use dfn_core::api;
+use ic_base_types::CanisterId;
+use rand::{
+    distributions::{Distribution, WeightedIndex},
+    rngs::StdRng,
+    seq::SliceRandom,
+    Rng, SeedableRng,
+};
+use random_traffic_test::*;
+use std::cell::{Cell, RefCell};
+use std::collections::BTreeMap;
+use std::ops::RangeInclusive;
+
+thread_local! {
+    /// Random number generator used for determining payload sizes et.al.
+    static RNG: RefCell<StdRng> = RefCell::new(StdRng::seed_from_u64(13));
+    /// Weight for making a reply used in a weighted binomial distribution.
+    static REPLY_WEIGHT: Cell<u32> = const { Cell::new(1) };
+    /// Weight for making a downstream call used in a weighted binomial distribution.
+    static CALL_WEIGHT: Cell<u32> = const { Cell::new(0) };
+    /// A configuration holding parameters for how to the canister should behave, such as the range
+    /// of payload bytes it should send.
+    static CONFIG: RefCell<Config> = RefCell::default();
+    /// The maximum number of calls each heartbeat can make.
+    static MAX_CALLS_PER_HEARTBEAT: Cell<u32> = Cell::default();
```

Comment:
It's not random, rather the heartbeat makes calls in quick succession until it either hits this limit or observes a synchronous rejection or both. This is spelled out in the doc comment for `heartbeat()`.

Reply:
Got that. Just that the name and the (previous) doc comment of this could easily be interpreted differently without first consulting the doc comment of `heartbeat()`.